### PR TITLE
Add `can_have_content` getter for Element

### PR DESCRIPTION
**Pull request by mwcz**
Here's an example implementation of item 3 from #160.

### CODE REVIEW

File: src/rewritable_units/element.rs [hunk 1]
```diff
@@ -46,22 +46,22 @@ pub struct Element<'r, 't> {
     end_tag_mutations: Option<Mutations>,
     modified_end_tag_name: Option<Bytes<'static>>,
     end_tag_handler: Option<EndTagHandler<'static>>,
-    can_have_content: bool,
+    el_can_have_content: bool,
```
Comment (orium) [inline src/rewritable_units/element.rs:49]: Given that this is a `Element` the `el_` doesn't add much.  I think the original name is better.

File: src/rewritable_units/element.rs [hunk 2]
```diff
@@ -133,6 +133,14 @@ impl<'r, 't> Element<'r, 't> {
         self.start_tag.self_closing()
     }
 
+    /// Whether the element can have inner content.  In other words, is it either a self-closing
```
Comment (orium) [inline src/rewritable_units/element.rs:136]: > In other words, is it either a self-closing tag like `<foo />` or is an HTML void element which are implicitly self-closing, e.g. ``.

This is the situation when the method returns false.  Can you clarify by saying something like "This returns `true` unless it is a self-closing tag ..." ?

File: src/rewritable_units/element.rs (hunk 1 above)
Comment (mwcz) [reply src/rewritable_units/element.rs:49]: I agree, but the field and the function can't have the same name.  Maybe the field could keep it's name and we just make it `pub` instead?

Comment (orium) [reply src/rewritable_units/element.rs:49]: > the field and the function can't have the same name

It can.  Give it a try :P

Comment (mwcz) [reply src/rewritable_units/element.rs]: Weird, I could have sworn I tried that first and the compiler didn't like it.  Fixed.

File: src/rewritable_units/element.rs (hunk 2 above)
Comment (mwcz) [reply src/rewritable_units/element.rs]: Changed to:

```
Whether the element can have inner content.  Returns `true` unless the element is an [HTML void
element](https://html.spec.whatwg.org/multipage/syntax.html#void-elements) or has a
self-closing tag (eg, `<foo />`).
```

File: src/rewritable_units/element.rs [hunk 3]
```diff
@@ -53,15 +53,15 @@ pub struct Element<'r, 't> {
 }
 
 impl<'r, 't> Element<'r, 't> {
-    pub(crate) fn new(start_tag: &'r mut StartTag<'t>, can_have_content: bool) -> Self {
+    pub(crate) fn new(start_tag: &'r mut StartTag<'t>, el_can_have_content: bool) -> Self {
```
Comment (orium) [inline src/rewritable_units/element.rs:56]: You can also remove the `el_` here.

Comment (mwcz) [reply src/rewritable_units/element.rs]: Good catch, LSP rename missed that one, as did I.

**Comment by orium** [comment]
Needs to land after https://github.com/cloudflare/lol-html/pull/165 since it fixes some warnings.

**Comment by orium** [comment]
Just landed #165 which should fix the warnings.  Can you rebase on master and squash your commits into a single one please?

**Review by orium — APPROVED** [review]

**Comment by orium** [comment]
Merged.  Thank you!

**Pull request merged**